src/trajectory.jl

```diff
@@ -688,18 +718,15 @@ function mh_accept_ratio(
     Hproposal::AbstractVector{<:T},
 ) where {T<:AbstractFloat}
     α = min.(one(T), exp.(Horiginal .- Hproposal))
-    accept = _rand(rng, length(Horiginal)) .< α
+    # NOTE: There is a chance that sharing the RNG over multiple
+    #       chains for accepting / rejecting might couple
+    #       the chains. We need to revisit this more rigirously 
+    #       in the future. See discussions at 
+    #       https://github.com/TuringLang/AdvancedHMC.jl/pull/166#pullrequestreview-367216534
+    accept = rand(rng, T, length(Horiginal)) .< α
```

Comment:
This is potentially confusing since `rnd` in the argument has a type `rng::Union{AbstractRNG, AbstractVector{<:AbstractRNG}}`. We can simplify by enforcing a constraint that the user has to pass a vector for vectorised HMC, and a scalar otherwise. Any reason why we want to keep support for `rng` being an `AbstractRNG` type in vectorised HMC?

Reply:
One reason is for to be compatible with `Random.seed!`, `Random.GLOBAL_RNG` is reset. In this case, `rng = GLOBAL_RNG` controls all randomness. Also I think it makes sense to control the randomness using a single RNG in general.

Reply:
> Also I think it makes sense to control the randomness using a single RNG in general.

Can we automatically generate random seeds for `n_chains` using a separate global RNG stream?  This way all randomness is controlled by a single global RNG, but still, keep RNG independent for separate chains.

Reply:
That's a good solution!

Reply:
I took a look into how to remove all places with `Union{AbstractRNG, AbstractVector{<:AbstractRNG}}` and some of them requires some works on the metrics space part. I think this is out of the scope of this PR and worth a seperate one because it actually influences the whole code space. I created an issue (https://github.com/TuringLang/AdvancedHMC.jl/issues/186) to track this.

Reply:
Sounds good to me.